migration_scripts/reddit_author_created.py

```diff
@@ -0,0 +1,124 @@
+"""Batch script for data migration on data source Reddit
+If the field author.created is -1 (this can be the author of the post itself, but also the author
+of a comment), fetch the information via the PRAW API and recalculate the field member_since based
+on author.created and insert_date / comment.created
+"""
+import datetime
+import praw
+from utils import read_args, get_database
+from constants import REDDIT_CLIENT_ID, REDDIT_CLIENT_SECRET, REDDIT_USER_AGENT
+
+if __name__ == '__main__':
+    remote = read_args(__file__)
+    database = get_database(remote)
+    collection = database['reddit.posts']
+
+    # Set up PRAW API client
+    api = praw.Reddit(client_id=REDDIT_CLIENT_ID, client_secret=REDDIT_CLIENT_SECRET, user_agent=REDDIT_USER_AGENT)
+
+    post_authors = collection.aggregate([
+        {
+            '$match': {
+                'author.created': -1
+            }
+        }, {
+            '$group': {
+                '_id': '$author.name'
+            }
+        }, {
+            '$limit': 10
```

Comment:
Commenting here, but the remark refers to all MongoDB queries in every script: Why are the results limited to 10 here? Then we'd have to execute this script approx. 30.000 times to update all entries in the datebase (for one script), don't we?